src/driver-hidpp20.c

```diff
@@ -1166,6 +1166,9 @@ hidpp20drv_commit(struct ratbag_device *device)
 		drv_data->profiles->profiles[profile->index].enabled = profile->is_enabled;
 
 		ratbag_profile_for_each_resolution(profile, resolution) {
+			if (!resolution->dirty)
+				continue;
+
```

Comment:
This test needs to be placed after `dpi_index = resolution->index;` or changing any other setting will switch the resolution to the first one (`dpi_index` will stay at `0`)